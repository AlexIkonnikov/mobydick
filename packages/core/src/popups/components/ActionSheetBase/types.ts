import {StyleProp, ViewStyle} from 'react-native';
import React, {PropsWithChildren, ReactElement} from 'react';

import {TypographyProp} from '../../../typography/types';
import {IPopup} from '../../types';

export interface IPropsContents {
  title: string;
  textFont?: TypographyProp;
  leftIcon?: ReactElement | undefined;
  onPress?(): void;
  radio?: string;
  checkboxList?: string[];
  disabled?: boolean;
  itemType: IItemType;
}

export enum IItemType {
  firstItem = 'firstItem',
  innerItem = 'innerItem',
  lastItem = 'lastItem',
  singleItem = 'singleItem',
  cancelItem = 'cancelItem',
}

export interface IPropsItem extends IPropsContents {
  style?: ViewStyle;
  isStatusPressedForTest?: boolean;
}

export interface IActionSheetBaseProps
  extends PropsWithChildren<Omit<IPopup, 'Content'>> {
  onClose: () => void;
  overlayStyle?: StyleProp<ViewStyle>;
  containerStyle?: StyleProp<ViewStyle>;
  children: React.ReactNode;
}
