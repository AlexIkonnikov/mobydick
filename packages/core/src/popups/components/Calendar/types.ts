export interface IChangeDate {
  dateStart: string;
  dateEnd: string;
}

export enum IButtonView {
  small = 'small',
  large = 'large',
}
