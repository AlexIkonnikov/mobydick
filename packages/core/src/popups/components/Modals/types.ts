import {IButtonTypes} from '../../../cta';

export interface IModalProps {
  title: string;
  descriptionText: string;
  buttonText?: string;
}

export interface IModalAskProps {
  title: string;
  descriptionText: string;

  typeLeft?: IButtonTypes;
  textLeft?: string;

  onPressRight(): void;
  typeRight?: IButtonTypes;
  textRight?: string;
}
