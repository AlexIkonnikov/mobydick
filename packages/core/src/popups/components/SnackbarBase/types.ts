import {StyleProp, ViewStyle} from 'react-native';
import {PropsWithChildren} from 'react';

import {IPopup, IPosition} from '../../types';

export interface ISnackbarProps
  extends PropsWithChildren<Omit<IPopup, 'Content'>> {
  onClose: () => void;
  containerStyle?: StyleProp<ViewStyle>;
  position: IPosition;
  timeShow?: number;
}
