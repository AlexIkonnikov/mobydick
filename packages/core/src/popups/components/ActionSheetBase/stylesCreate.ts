import {ViewStyle} from 'react-native';

import rem from '../../../styles/utils/rem';
import {createStyles} from '../../../styles';

import {IItemType} from './types';

const stylesCreate = createStyles(({spaces, colors}, itemType?: IItemType) => {
  const getItemStyle = (): ViewStyle => {
    switch (itemType) {
      case IItemType.firstItem:
        return {
          borderTopLeftRadius: spaces.Space12,
          borderTopRightRadius: spaces.Space12,
        };
      case IItemType.innerItem:
        return {
          borderTopWidth: spaces.Space1,
          borderTopColor: colors.BgTertiary,
        };
      case IItemType.lastItem:
        return {
          borderBottomLeftRadius: spaces.Space12,
          borderBottomRightRadius: spaces.Space12,
          marginBottom: spaces.Space8,
          borderTopWidth: spaces.Space1,
          borderTopColor: colors.BgTertiary,
        };
      case IItemType.cancelItem:
        return {
          borderRadius: spaces.Space12,
          marginBottom: rem(30),
          justifyContent: 'center',
        };
      default:
        return {
          borderRadius: spaces.Space12,
          marginBottom: spaces.Space8,
          justifyContent: 'center',
        };
    }
  };

  return {
    overlayStyle: {
      justifyContent: 'flex-end',
    },
    containerStyle: {
      width: '100%',
    },
    item: {
      justifyContent: 'space-between',
      flexDirection: 'row',
      alignItems: 'center',
      paddingHorizontal: spaces.Space20,
      marginHorizontal: spaces.Space8,
      minHeight: rem(50),

      ...getItemStyle(),
    },
    leftIcon: {
      paddingRight: spaces.Space12,
    },
    leftIconView: {
      justifyContent: 'flex-start',
      flexDirection: 'row',
      alignItems: 'center',
    },
    textSelected: {
      paddingVertical: rem(15),
    },
  };
});

export default stylesCreate;
