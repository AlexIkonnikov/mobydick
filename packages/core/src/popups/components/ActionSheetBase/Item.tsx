import {FC, useCallback} from 'react';

import useStyles from '../../../styles/hooks/useStyles';
import useTheme from '../../../styles/hooks/useTheme';
import Pressable from '../../../basic/components/Pressable/Pressable';
import {LABELS} from '../../../other';

import stylesCreate from './stylesCreate';
import Contents from './content/Contents';
import {IPropsItem} from './types';

const Item: FC<IPropsItem> = props => {
  const {onPress, style, disabled, itemType, isStatusPressedForTest} = props;
  const [styles] = useStyles(stylesCreate, itemType);
  const {colors} = useTheme();

  const getStyle = useCallback(
    ({pressed}: {pressed: boolean}) => [
      styles.item,
      {backgroundColor: pressed ? colors.BgSecondary : colors.BgPrimary},
      style,
    ],
    [],
  );

  return (
    <Pressable
      style={getStyle}
      disabled={disabled}
      onPress={onPress}
      accessibilityLabel={LABELS.actionSheetsItem}
      testOnly_pressed={isStatusPressedForTest}>
      <Contents {...props} />
    </Pressable>
  );
};

export default Item;
