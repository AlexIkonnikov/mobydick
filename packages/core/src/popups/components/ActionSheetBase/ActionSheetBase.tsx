import {FC} from 'react';

import {PopupBase} from '../PopupBase';
import {returnTrue} from '../../functions';
import useStyles from '../../../styles/theme/hooks/useStyles';
import View from '../../../basic/components/View/View';

import stylesCreate from './stylesCreate';
import Item from './Item';
import {IActionSheetBaseProps} from './types';

const ActionSheetBase: FC<IActionSheetBaseProps> & {
  Item: typeof Item;
} = props => {
  const {children, overlayStyle, onClose, containerStyle} = props;
  const [styles] = useStyles(stylesCreate);

  return (
    <PopupBase
      onClose={onClose}
      overlayStyle={[styles.overlayStyle, overlayStyle]}>
      <View
        style={[styles.containerStyle, containerStyle]}
        onStartShouldSetResponder={returnTrue}>
        {children}
      </View>
    </PopupBase>
  );
};

ActionSheetBase.Item = Item;
export default ActionSheetBase;
