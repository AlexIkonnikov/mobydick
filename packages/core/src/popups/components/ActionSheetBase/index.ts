import ActionSheetBase from './ActionSheetBase';

export {ActionSheetBase};

export * from './types';
