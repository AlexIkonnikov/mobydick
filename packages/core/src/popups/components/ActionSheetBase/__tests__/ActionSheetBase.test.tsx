import {render} from '@testing-library/react-native';

import ActionSheetBase from '../ActionSheetBase';
import {IItemType} from '../types';

describe('@lad-tech/mobydick-core/ActionSheetBase', () => {
  it('should renders correctly', () => {
    const {toJSON} = render(
      <ActionSheetBase id={'id'} onClose={() => null}>
        <></>
      </ActionSheetBase>,
    );

    expect(toJSON()).toMatchSnapshot();
  });
  it('should renders correctly with item', () => {
    const {toJSON} = render(
      <ActionSheetBase id={'id'} onClose={() => null}>
        <ActionSheetBase.Item
          textFont={'Regular-White-S'}
          itemType={IItemType.innerItem}
          title={'title'}
        />
      </ActionSheetBase>,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
