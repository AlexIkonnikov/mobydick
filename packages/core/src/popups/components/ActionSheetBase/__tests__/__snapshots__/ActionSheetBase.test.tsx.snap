// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@lad-tech/mobydick-core/ActionSheetBase should renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.2)",
        "height": "100%",
        "position": "absolute",
        "width": "100%",
      },
      [
        {
          "justifyContent": "flex-end",
        },
        undefined,
      ],
    ]
  }
  testID="Pressable"
>
  <View
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "width": "100%",
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`@lad-tech/mobydick-core/ActionSheetBase should renders correctly with item 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.2)",
        "height": "100%",
        "position": "absolute",
        "width": "100%",
      },
      [
        {
          "justifyContent": "flex-end",
        },
        undefined,
      ],
    ]
  }
  testID="Pressable"
>
  <View
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityLabel="actionSheetsItem"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "alignItems": "center",
            "borderTopColor": "#D1D4DB",
            "borderTopWidth": 1.5,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginHorizontal": 12,
            "minHeight": 75,
            "paddingHorizontal": 30,
          },
          {
            "backgroundColor": "#FFF",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <Text
          allowFontScaling={true}
          style={
            [
              {
                "color": "#FFF",
                "fontFamily": "Inter-Regular",
                "fontSize": 22.5,
                "lineHeight": 28.5,
              },
              {
                "paddingVertical": 22.5,
              },
            ]
          }
        >
          title
        </Text>
      </View>
    </View>
  </View>
</View>
`;
